### PR TITLE
feat(core/grid): add grid component

### PR DESCRIPTION
**Pull request by danielleroux**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`yarn build`) was run locally and any changes were pushed
- [ ] Unit tests (`yarn test`) were run locally and passed
- [ ] Visual Regression Tests (`yarn visual-regression`) were run locally and passed
- [ ] Linting (`npm lint`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bug fix
- [x] Feature
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/index.html [hunk 1]
```diff
@@ -29,7 +29,7 @@
     </script>
   </head>
 
-  <body style="margin: 0px; height: 100vh" class="theme-classic-dark">
+  <body style="margin: 0px; height: 100vh" class="theme-brand-dark">
```
Comment (nuke-ellington) [inline packages/core/src/index.html:32]: on purpose?

File: packages/core/src/tests/basic-navigation/basic-navigation.e2e.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ regressionTest.describe('basic navigation', () => {
     await page.goto('basic-navigation/basic');
     await page.setViewportSize({
       height: 1200,
-      width: mediumWidth,
+      width: 780,
```
Comment (nuke-ellington) [inline packages/core/src/tests/basic-navigation/basic-navigation.e2e.ts:55]:
```suggestion
      width: mediumWidth,
```

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

File: packages/core/src/components/grid/grid.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+ * @since 2.0.0
+ */
+@Component({
+  tag: 'ix-grid',
+  styleUrl: 'grid.scss',
+  shadow: true,
+})
+export class Grid {
+  /**
+   * Grid will be displayed without any padding
```
[the remark this answers is not in the document]
Comment (danielleroux) [reply packages/core/src/components/grid/grid.tsx:29]: Wrong jsdocs here

File: packages/core/src/components/grid/grid.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+ * @since 2.0.0
+ */
+@Component({
+  tag: 'ix-grid',
+  styleUrl: 'grid.scss',
+  shadow: true,
+})
+export class Grid {
+  /**
+   * Grid will displayed with a fixed width
```
Comment (nuke-ellington) [inline packages/core/src/components/grid/grid.tsx:29]:
```suggestion
   * The grid will have a fixed width
```

**Review by nuke-ellington — CHANGES_REQUESTED** [review]
One small remark - sorry :)

**Review by nuke-ellington — APPROVED** [review]
lgtm

**Pull request merged**